packages/did/src/Did.chain.ts

```diff
@@ -403,14 +401,70 @@ export async function generateCreateTxFromDidDetails(
   const signature = await signer.sign({
     data: encodedDidCreationDetails.toU8a(),
     meta: {},
-    publicKey: Crypto.coToUInt8(signingPublicKey),
-    alg,
+    publicKey: Crypto.coToUInt8(authenticationKey.publicKey),
+    alg: getSignatureAlgForKeyType(authenticationKey.type) as string,
   })
   return api.tx.did.create(encodedDidCreationDetails, {
     [signature.alg]: signature.data,
   })
 }
 
+export async function generateCreateTxFromDidDetails(
+  did: DidDetails,
+  submitterAddress: IIdentity['address'],
+  signer: KeystoreSigner
+): Promise<SubmittableExtrinsic> {
+  const { authenticationKey } = did
+  if (!authenticationKey) {
+    throw SDKErrors.ERROR_DID_ERROR(
+      `The provided DID does not have an authentication key to sign the creation operation.`
+    )
+  }
+
+  const keyAgreementKeys = did.getKeys(
+    KeyRelationship.keyAgreement
+  ) as DidEncryptionKey[]
```

Comment:
C: every time you use `as` you are circumventing typescript and force a type. Not sure why we don't have the eslint rule for that anymore, but I think, having y use `as` is rather a sign, that something is not optimal.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/473/commits/5234b12f7bc5848b609e38cd75eea91aaf4ffd2e